upf/solvers/plan_validator.py

```diff
@@ -167,15 +162,15 @@ def walk_param_exp(self, expression: FNode, args: List[FNode]) -> FNode:
             raise UPFProblemDefinitionError(f"Value of ActionParameter {str(expression)} not found in {str(self._assignments)}")
 
 
-class SequentialPlanValidator(upf.Solver):
+class SequentialPlanValidator(solvers.solver.Solver):
     """Performs plan validation."""
     def __init__(self, **options):
-        self._env: 'upf.environment.Environment' = upf.get_env(options.get('env', None))
+        self._env: 'upf.environment.Environment' = upf.environment.get_env(options.get('env', None))
         self.manager = self._env.expression_manager
-        self._substituter = Substituter(self._env)
+        self._substituter = walkers.Substituter(self._env)
         self._last_error: Union[str, None] = None
 
-    def validate(self, problem: 'upf.Problem', plan: 'upf.Plan') -> bool:
+    def validate(self, problem: 'upf.model.problem.Problem', plan: 'upf.plan.Plan') -> bool:
```

Comment:
```suggestion
    def validate(self, problem: 'upf.model.Problem', plan: 'upf.plan.Plan') -> bool:
```
or
```suggestion
    def validate(self, problem: 'Problem', plan: 'upf.plan.Plan') -> bool:
```

The second works because you import `Problem` at line 24.